libs/langchain-google-cloud-sql-pg/src/vectorStore.ts

```diff
@@ -0,0 +1,315 @@
+      k = 4,
+      fetch_k = 20,
+      lambdaMult = 0.5,
+      indexQueryOptions
+    }: PostgresVectorStoreArgs
+  ): Promise<PostgresVectorStore> {
+
+    if(metadataColumns !== undefined && ignoreMetadataColumns !== undefined) {
+      throw "Can not use both metadata_columns and ignore_metadata_columns.";
+    }
+
+    const {rows} = await engine.pool.raw(`SELECT column_name, data_type FROM information_schema.columns WHERE table_name = '${tableName}' AND table_schema = '${schemaName}'`);
+    let columns: {[key: string]: any} = {};
+
+    for (const index in rows) {
+      const row = rows[index];
+      columns[row["column_name"]] = row["data_type"]
+    }
+
+    if(!columns.hasOwnProperty(idColumn)){
+      throw `Id column: ${idColumn}, does not exist.`;
+    }
+
+    if(!columns.hasOwnProperty(contentColumn)){
+      throw `Content column: ${contentColumn}, does not exist.`;
+    }
+
+    const contentType = columns[contentColumn];
+
+    if(contentType !== "text" && !contentType.includes("char")) {
+      throw `Content column: ${contentColumn}, is type: ${contentType}. It must be a type of character string.`
+    }
+
+    if(!columns.hasOwnProperty(embeddingColumn)) {
+      throw `Embedding column: ${embeddingColumn}, does not exist.`
+    }
+        
+    if(columns[embeddingColumn] !== "USER-DEFINED") {
+      throw `Embedding column: ${embeddingColumn} is not of type Vector.`
+    }
+
+    metadataJsonColumn = columns.hasOwnProperty(metadataJsonColumn) ? metadataJsonColumn : "";
+
+    for (const column of metadataColumns) {
+      if(!columns.hasOwnProperty(column)) {
+        throw `Metadata column: ${column}, does not exist.`
+      }
+    }
+
+    const allColumns = columns;
+    if(ignoreMetadataColumns !== undefined && ignoreMetadataColumns.length > 0) {
+      for (const column of ignoreMetadataColumns) {
+        delete allColumns[column];
+      }
+
+      delete allColumns[idColumn];
+      delete allColumns[contentColumn];
+      delete allColumns[embeddingColumn];
+      metadataColumns = Object.keys(allColumns);
+    } 
+    return new PostgresVectorStore(
+        PostgresVectorStore._createKey,
+        engine,
+        embeddings,
+        tableName,
+        schemaName,
+        contentColumn,
+        embeddingColumn,
+        metadataColumns,
+        ignoreMetadataColumns!,
+        idColumn,
+        metadataJsonColumn,
+        distance_strategy,
+        k,
+        fetch_k,
+        lambdaMult,
+        indexQueryOptions!
+      )
+  }
+
+  async addEmbeddings(
+    texts: Iterable<string>,
+    embeddings: number[][], 
+    metadatas?: Array<Record<string, any>> | null,
+    ids?: { [x: string]: any; },
+  ): Promise<{[x: string]: any;}> {
+
+    if(!ids) {
+      ids = Array.from(texts).map(() => uuidv4());
+    }
+    if(!metadatas) {
+      metadatas = Array.from(texts).map(() => new Object());
+    }
+
+    const tuples = customZip(ids, texts, embeddings, metadatas)
```

Comment:
I need to review this closer, but I think you need to be using the pgvector-node library https://github.com/pgvector/pgvector-node to handle the vectors appropriately.